R/splatPopParams-methods.R

```diff
@@ -0,0 +1,106 @@
+                eqtl.ES.shape = checkNumber(v$eqtl.ES.shape, lower = 0),
+                eqtl.ES.rate = checkNumber(v$eqtl.ES.rate, lower = 0),
+                eqtl.group.specific =checkNumber(v$eqtl.group.specific, 
+                                                 lower = 0, upper = 1),
+                pop.mean.shape = checkNumber(v$pop.mean.shape, lower = 0),
+                pop.mean.rate = checkNumber(v$pop.mean.rate, lower = 0),
+                pop.cv.bins = checkInt(v$pop.cv.bins, lower=1),
+                pop.cv.param = checkDataFrame(v$pop.cv.param))
+    
+    if (all(checks == TRUE)) {
+        valid <- TRUE
+    } else {
+        valid <- checks[checks != TRUE]
+        valid <- paste(names(valid), valid, sep = ": ")
+    }
+    
+    return(valid)
+})
+
+
+#' @importFrom methods callNextMethod
+setMethod("show", "splatPopParams", function(object) {
+    
+    pp <- list("Population params:" = c("[random.genes]" = "random.genes",
+                                        "(mean.shape)" = "pop.mean.shape",
+                                        "(mean.rate)" = "pop.mean.rate",
+                                        "[group.prop]" = "group.prop",
+                                        "[cv.bins]" = "pop.cv.bins",
+                                        "(cv.params)" = "pop.cv.param"),
+               "eQTL params:" = c("[eqtl.n]"    = "eqtl.n",
+                                  "[eqtl.dist]" = "eqtl.dist",
+                                  "[eqtl.maf.min]" = "eqtl.maf.min",
+                                  "[eqtl.maf.max]" = "eqtl.maf.max",
+                                  "[eqtl.group.specific]" = "eqtl.group.specific",
+                                  "(eqtl.ES.shape)" = "eqtl.ES.shape",
+                                  "(eqtl.ES.rate)" = "eqtl.ES.rate"))
+    
+    callNextMethod()
+    showPP(object, pp)
+})
+
+
+#' @rdname setParam
+setMethod("setParam", "splatPopParams", function(object, name, value) {
+    checkmate::assertString(name)
+    
+    if (name == "nCells") {
+        warning(name, " only used if genes='random'")
+    }
+    
+    if (name == "pop.cv.param") {
+        if (getParam(object, "pop.cv.bins") != nrow(value)) {
+            stop("Need to set pop.cv.bins to length of pop.cv.param")
+        }
+    }
+    
+    if (name == "eqtl.maf.min") {
+        if (getParam(object, "eqtl.maf.min") >= getParam(object, "eqtl.maf.max")) {
+            stop("Range of acceptable Minor Allele Frequencies is too small...
+                 Be sure eqtl.maf.min < eqtl.maf.max.")
+        }
+    }
+    
+    object <- callNextMethod()
+    
+    return(object)
+})
+
+#' @rdname setParams
+setMethod("setParams", "splatPopParams", function(object, update = NULL, ...) {
+    
+    checkmate::assertClass(object, classes = "splatPopParams")
+    checkmate::assertList(update, null.ok = TRUE)
+    
+    update <- c(update, list(...))
+    
+    object <- callNextMethod(object, update)
+    
+    return(object)
+})
```

Comment:
There's nothing specific to `SplatPopParams` here so I think we can drop it and just use the `SplatParams` method.

Reply:
does that mean modifying setMethod in SplatParams-methods.R to: 
setMethod("setParams", c("splatPop", "splatPopParams"), function(object, update = NULL, ...)

Reply:
No I think you should just be able to delete this and inheritance should fall back to the `SplatParams` method without any modifications. I think so anyway... Should be easy enough to test by commenting this out, loading the package and seeing if `setParams()` still works.

Reply:
That worked so I will remove setParams from splatPopParams-methods.